pkg/services/dashboards/service/zanzana.go

```diff
@@ -85,13 +90,19 @@ func (dr *DashboardServiceImpl) findDashboardsZanzanaCompare(ctx context.Context
 	return first.result, first.err
 }
 
+type checkDashboardsFn func(context.Context, dashboards.FindPersistedDashboardsQuery, []dashboards.DashboardSearchProjection, int64) ([]dashboards.DashboardSearchProjection, error)
+
 func (dr *DashboardServiceImpl) findDashboardsZanzana(ctx context.Context, query dashboards.FindPersistedDashboardsQuery) ([]dashboards.DashboardSearchProjection, error) {
-	return dr.findDashboardsZanzanaCheck(ctx, query)
+	if len(query.Title) <= listQueryLengthThreshold || query.Limit > listQueryLimitThreshold {
+		return dr.findDashboardsZanzanaCheck(ctx, query, dr.checkDashboardsCompile)
```

Comment:
So for this one we only have to call compile for each resource once, right now we call it every time we have new results right?

Reply:
What I am trying to say is that we should be able to construct the checkers once and re-use them when we need to re fetch more data from the db